deployments/constructs/platforms/azure-platform.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2023 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository https://github.com/carbynestack/carbynestack.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as azurerm from "@cdktf/provider-azurerm";
+import * as kubernetes from "@cdktf/provider-kubernetes";
+import * as kubectl from "../../.gen/providers/kubectl";
+import * as helm from "@cdktf/provider-helm";
+import { Istio } from "../backing-services/istio";
+import { Fn } from "cdktf";
+import { Construct } from "constructs";
+
+export interface AKSConfig {
+  resourceGroup: string | azurerm.resourceGroup.ResourceGroup;
+  virtualNetworkAddressSpace?: string[];
+  subnetAddressPrefixes: string[];
+  location: string;
+  privateDnsZone?: azurerm.privateDnsZone.PrivateDnsZone;
+}
+
+export class AzurePlatform extends Construct {
+  public ingressIp: string;
+  public kubernetesProvider: kubernetes.provider.KubernetesProvider;
+  public kubectlProvider: kubectl.provider.KubectlProvider;
+  public helmProvider: helm.provider.HelmProvider;
+  public virtualNetwork: azurerm.virtualNetwork.VirtualNetwork;
+  public resourceGroup: azurerm.resourceGroup.ResourceGroup;
+
+  constructor(scope: Construct, name: string, config: AKSConfig) {
+    super(scope, name);
+
+    // if the resource group is a string, create a new resource group with the provided name, otherwise use the provided resource group
+    if (typeof config.resourceGroup === "string") {
+      this.resourceGroup = new azurerm.resourceGroup.ResourceGroup(
+        this,
+        `${name}-rg`,
+        {
+          name: config.resourceGroup,
+          location: config.location,
+        },
+      );
+    } else {
+      this.resourceGroup = config.resourceGroup;
+    }
+
+    this.virtualNetwork = new azurerm.virtualNetwork.VirtualNetwork(
+      this,
+      `${name}-vnet`,
+      {
+        name: `${name}-vnet`,
+        resourceGroupName: this.resourceGroup.name,
+        location: this.resourceGroup.location,
+        addressSpace: config.virtualNetworkAddressSpace!,
+      },
+    );
+
+    // subnet
+    const subnet = new azurerm.subnet.Subnet(this, `${name}-subnet`, {
+      name: `${this.resourceGroup.name}-subnet`,
+      resourceGroupName: this.resourceGroup.name,
+      virtualNetworkName: this.virtualNetwork.name,
+      addressPrefixes: config.subnetAddressPrefixes,
+    });
+
+    const kubernetesCluster = new azurerm.kubernetesCluster.KubernetesCluster(
+      this,
+      `${name}-aks`,
+      {
+        name: `${name}-aks`,
+        location: config.location,
+        resourceGroupName: this.resourceGroup.name,
+        dnsPrefix: "dns",
+        defaultNodePool: {
+          name: "agentpool",
+          nodeCount: 1,
+          vmSize: "Standard_D3_v2",
```

Comment:
can this be made configurable using a config file or flag?

Reply:
Would it make sense?

Reply:
it can now be passed via optional configuration values with the current values now as default.